### ISSUE TITLE
Rails/Delegate ignores ruby 2.3 Safe Nagivation Operator

### ISSUE THREAD

**Issue by PizzaPill** [bbatsov/rubocop#3040]
The following line of code:

`object_that_can_be_nil&.foo` gets converted to `delegate :foo, to: :object_that_can_be_nil, prefix: true` which is not equivalent. The equivalent would be `delegate :foo, to: :object_that_can_be_nil, prefix: true, allow_nil: true`. 

This all happens in rubocop 0.39.0 and parameter `auto-correct` enabled.

**Comment by jonas054** [comment]
@geniou You created that cop. Can you look into this?

**Comment by andialles** [comment]
Same thing happens here
```
def foo
    bar.foo if bar
end
```
gets converted to `delegate :foo, to: :bar` without the `allow_nil: true` parameter

**Comment by cgriego** [comment]
One gotcha here is that `allow_nil` is not equivalent to `&.` because `allow_nil` checks that the receiver is not nil or if it responds to the method. So methods that nil responds to are also delegated. This is problematic, but the only handy example I have is with whiny nils.

```ruby
# nil if foo is nil
def foo_id
  foo&.id
end

# RuntimeError if foo is nil and whiny nils
delegate :id, to: :foo, prefix: true, allow_nil: true
```
